client/session.go

```diff
@@ -871,10 +875,19 @@ func (s *session) fetchAllAttempt(
 		resultErrs             int32
 		majority               int32
 		fetchBatchOpsByHostIdx [][]*fetchBatchOp
-		nsID                   = []byte(namespace)
 		success                = false
 	)
 
+	// NB(prateek): need to make a copy of inputNamespace and inputIDs to control
+	// their life-cycle within this function.
+	namespace := s.idPool.Clone(inputNamespace)
+	idsClone := inputIDs.Clone()
+	ids := s.idPool.CloneIDs(idsClone)
```

Comment:
Strange, what are we cloning 2x here?  The slice and then the IDs or something like that?  

Any reason the idPool wouldn't do the slice cloning?

Reply:
`idsClone := inputIDs.Clone()` clones the iterator (only the struct referencing the underlying slice, not the slice itself). Need this to be able to iterate the original iterator multiple times in case of retries.

`ids := s.idPool.CloneIDs(..)` actually clones the ids in the slice.

Reply:
Right I see, makes sense.